pkg/cpuallocator/allocator.go

```diff
@@ -438,54 +438,56 @@ func (c *topologyCache) discoverSstCPUPriority(sys sysfs.System, pkgID idset.ID)
 	cpuIDs := c.pkg[pkgID].ToSlice()
 	prios := make(map[idset.ID]CPUPriority, len(cpuIDs))
 
-	// Determine SST-based priority. Based on experimentation there is some
-	// hierarchy between the SST features. Without trying to be too smart
-	// we follow the principles below:
-	// 1. SST-TF has highest preference, mastering over SST-BF and making most
-	//    of SST-CP settings ineffective
-	// 2. SST-CP dictates over SST-BF
-	// 3. SST-BF is meaningful if neither SST-TF nor SST-CP is enabled
-	switch {
-	case sst.TFEnabled:
-		log.Debug("package #%d: using SST-TF based CPU prioritization", pkgID)
-		// We only look at the CLOS id as SST-TF (seems to) follows ordered CLOS priority
-		for _, i := range cpuIDs {
-			id := idset.ID(i)
-			p := PriorityLow
-			// First two CLOSes are prioritized by SST
-			if sys.CPU(id).SstClos() < 2 {
-				p = PriorityHigh
+	if sst != nil {
+		// Determine SST-based priority. Based on experimentation there is some
+		// hierarchy between the SST features. Without trying to be too smart
+		// we follow the principles below:
+		// 1. SST-TF has highest preference, mastering over SST-BF and making most
+		//    of SST-CP settings ineffective
+		// 2. SST-CP dictates over SST-BF
+		// 3. SST-BF is meaningful if neither SST-TF nor SST-CP is enabled
+		switch {
```

Comment:
nitpick: since you have an expressionless switch, I'd suggest making `case sst == nil:` the first branch/case.